### PR TITLE
Absolute paths in typedefs, fix for pyang 1.7.1

### PR DESCRIPTION
**Pull request by robshakir**
```
  * (M) openconfig_pyang/plugins/openconfig.py
    - Ensure that the enumerated error types that are created in
      this module can be compared by pyang 1.7.1.
    - Don't throw the warning for absolute paths where the object
      being examined is a typedef.
```

### CODE REVIEW

File: openconfig_pyang/plugins/openconfig.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 from __future__ import print_function
 
-from enum import Enum
+from enum import IntEnum, Enum
```
Comment (aashaikh) [inline openconfig_pyang/plugins/openconfig.py:24]: is `enum` a standard library package?   it tripped me up when using on new systems

File: tests/oclinter/key-quoting-submodule/openconfig-submodule-fail.yang [hunk 2]
```diff
@@ -0,0 +1,62 @@
+submodule openconfig-submodule-fail {
+  belongs-to openconfig-testcase-fail {
+    prefix "oc-tc";
+  }
+
+  import openconfig-extensions { prefix oc-ext; }
+
+  description
+    "Success testcase to check compliance with quoting
```
Comment (aashaikh) [inline tests/oclinter/key-quoting-submodule/openconfig-submodule-fail.yang]: failure testcase ?

File: tests/oclinter/key-quoting-submodule/openconfig-testcase-fail.yang [hunk 3]
```diff
@@ -0,0 +1,23 @@
+module openconfig-testcase-fail {
+  prefix "oc-tc";
+  namespace "http://openconfig.net/linter/testcase";
+
+  import openconfig-extensions { prefix oc-ext; }
+
+  include openconfig-submodule-fail;
+
+  description
+    "Success testcase to check compliance with quoting
```
Comment (aashaikh) [inline tests/oclinter/key-quoting-submodule/openconfig-testcase-fail.yang]: failure testcase

File: tests/oclinter/key-quoting/openconfig-testcase-succeed.yang [hunk 4]
```diff
@@ -29,7 +29,8 @@ module openconfig-testcase-succeed {
 
   grouping testcase-top {
     description
-      "Top-level grouping";
+      "Top-level grouping that uses the 
+      word key in it";
```
Comment (aashaikh) [inline tests/oclinter/key-quoting/openconfig-testcase-succeed.yang]: do you mean grouping that uses a list with a key statement ?

**Review by aashaikh — APPROVED** [review]
LGTM.  just some minor comments.

File: openconfig_pyang/plugins/openconfig.py (hunk 1 above)
Comment (robshakir) [reply openconfig_pyang/plugins/openconfig.py:24]: Updated to enum34, it's not a standard package. It's listed in `requirements.txt`.

File: tests/oclinter/key-quoting-submodule/openconfig-submodule-fail.yang (hunk 2 above)
Comment (robshakir) [reply tests/oclinter/key-quoting-submodule/openconfig-submodule-fail.yang]: Done.

File: tests/oclinter/key-quoting-submodule/openconfig-testcase-fail.yang (hunk 3 above)
Comment (robshakir) [reply tests/oclinter/key-quoting-submodule/openconfig-testcase-fail.yang]: Done.

File: tests/oclinter/key-quoting/openconfig-testcase-succeed.yang (hunk 4 above)
Comment (robshakir) [reply tests/oclinter/key-quoting/openconfig-testcase-succeed.yang]: No, it's to catch the case that we have the word key that is not within a key statement. It's to check that we don't error in that case.

**Pull request merged**